DatadogInternal/Sources/MessageBus/NewFeatureBaggage.swift

```diff
@@ -0,0 +1,69 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// A `FeatureBaggage` holds any codable value.
+///
+/// Values are uniquely identified by key as `String`, the value type is validated on `get`
+/// either explicity, when specified, or inferred.
+///
+/// ## Creates a Feature Baggage
+/// Create a baggage by providing an `Encodable` value.
+///
+/// The example below shows how to create a baggage from an instance of a simple `GroceryProduct`:
+///
+///     struct GroceryProduct: Encodable {
+///         var name: String
+///         var points: Int
+///         var description: String?
+///     }
+///
+///     let pear = GroceryProduct(name: "Pear", points: 250, description: "A ripe pear.")
+///     let baggage = try FeatureBaggage(pear)
+///
+/// The `pear` is stored as a dictionary within the baggage:
+///
+///     print(baggage.rawValue)
+///     // ["description": Optional("A ripe pear."), "points": Optional(250), "name": Optional("Pear")]
+///
+/// ## Accessing Value
+/// The baggage value can then be decoded to any type that follow the same schema.
+///
+/// The following example decodes the baggage into a new data type:
+///
+///     struct CartItem: Decodable {
+///         var name: String
+///         var points: Int
+///     }
+///
+///     let item: CartItem = try baggage.decode()
+///
+///     print(item)
+///     // CartItem(name: "Pear", points: 250)
+///
+/// A Feature Baggage does not ensure thread-safety of values that holds references, make
+/// sure that any value can be accessibe from any thread.
+public struct NewFeatureBaggage {
```

Comment:
Is the `New*` prefix intentional? 🤔💭 Aren't we replacing the previous concept entirely in this PR?

Reply:
Indeed, it will be replaced entirely in #1442
This PR applies the new baggage on the message-bus only, the second PR is about the core context.